src/lib/autoDetect.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  ExtensionContext,
+  workspace,
+  window,
+  ConfigurationTarget,
+  WorkspaceConfiguration,
+} from 'vscode';
+import { FileAssociations } from './types';
+const { extname } = require('path');
+import {
+  EXTENSION_CONFIG_NAME,
+  EXTENSION_CONFIG_KEYS,
+  HUBL_HTML_ID,
+  HUBL_CSS_ID,
+} from './constants';
+
+export const initializeHubLAutoDetect = (context: ExtensionContext) => {
+  if (
+    hasHubLAssociations() ||
+    checkGlobalState() ||
+    !!context.workspaceState.get('DO_NOT_USE_HUBL')
+  ) {
+    return;
+  }
+  const showHublDetectedMessage = () => {
+    const hublDetectedMessage = `It looks like this file contains HubL code. Would you like to use HubL for this project?`;
+    const hublDetectedYesButton = `Use HubL`;
+    const hublDetectedNoButton = 'No';
+    const hublDetectedNeverAgainButton = 'Never ask again';
+
+    window
+      .showInformationMessage(
+        hublDetectedMessage,
+        hublDetectedYesButton,
+        hublDetectedNoButton,
+        hublDetectedNeverAgainButton
+      )
+      .then((selection) => {
+        switch (selection) {
```

Comment:
Need to make sure we `break;` after these `case` statements; currently clicking yes will also enable the workspaceState flag and the global setting.